base/twiceprecision.jl

```diff
@@ -237,7 +237,7 @@ eltype(::Type{TwicePrecision{T}}) where {T} = T
 
 promote_rule(::Type{TwicePrecision{R}}, ::Type{TwicePrecision{S}}) where {R,S} =
     TwicePrecision{promote_type(R,S)}
-promote_rule(::Type{TwicePrecision{R}}, ::Type{S}) where {R,S} =
+promote_rule(::Type{TwicePrecision{R}}, ::Type{S}) where {R,S<:Number} =
```

Comment:
I'm not sure this is right. For some reason `TwicePrecision{T}`'s parameter does not need to be a Number. @timholy Why is that?